fdbserver/KeyValueStoreShardedRocksDB.actor.cpp

```diff
@@ -2012,11 +2075,12 @@ struct ShardedRocksDBKeyValueStore : IKeyValueStore {
 				return;
 			}
 
-			for (auto shard : *(a.dirtyShards)) {
-				shard->readIterPool->update();
+			if (SERVER_KNOBS->ROCKSDB_READ_RANGE_REUSE_ITERATORS) {
+				for (auto shard : *(a.dirtyShards)) {
+					shard->readIterPool->update();
+				}
 			}
 
-			a.done.send(Void());
```

Comment:
What's the reason to move this line?

Reply:
We're blocking future commits. We don't want to accept more commits before the flush finished.